phylanx/execution_tree/compiler/compiler.hpp

```diff
@@ -269,8 +292,12 @@ namespace phylanx { namespace execution_tree { namespace compiler
         {}
 
         function compose(std::list<function> && elements,
-            std::string const& name, std::string const& codename) const
+            primitive_name_parts name_parts, std::string const& codename) const
         {
+            // The compiler does not know if it was the name of the instance or
+            // a primitive.
+            name_parts.instance = std::move(name_parts.primitive);
```

Comment:
What happens to the `name_parts.primitive` member here? Will it get re-assigned?